docs/design/seccomp.md

```diff
@@ -0,0 +1,299 @@
+	GreaterThanOrEqualTo SeccompOperator = "greaterThanOrEqualTo"
+	LessThan             SeccompOperator = "lessThan"
+	LessThanOrEqualTo    SeccompOperator = "lassThanOrEqualTo"
+	MaskEqualTo          SeccompOperator = "maskEqualTo"
+)
+
+// Represents a specific syscall argument in a seccomp rule.
+type SeccompArg struct {
+	Index    uint            `json:"index"`
+	Value    uint64          `json:"value"`
+	ValueTwo uint64          `json:"valueTwo"`
+	Op       SeccompOperator `json:"op"`
+}
+```
+
+Along with this API change, there will be supporting code changes:
+
+1.  New REST API implementation
+2.  Client code
+2.  Validations/conversions
+3.  `kubectl` support
+
+### Pod API changes
+
+The container `securityContext` must be changed to allow specifying a seccomp profile:
+
+```go
+// SecurityContext holds security configuration that will be applied to a container.
+// Some fields are present in both SecurityContext and PodSecurityContext.  When both
+// are set, the values in SecurityContext take precedence.
+type SecurityContext struct {
+  // other fields omitted
+  SeccompProfileName `json:"seccompProfileName,omitempty"`
+}
+```
+
+The kubelet docker runtime implementation in `pkg/kubelet/dockertools` will
+require code changes to implement setting the seccomp profile when a container
+is run.  This API change should not affect the kubelet itself.
+
+### Pod Security Policy changes
+
+The `PodSecurityPolicy` type should be changed to control seccomp profiles:
+
+```go
+// PodSecurityPolicySpec defines the policy enforced.
+type PodSecurityPolicySpec struct {
+  // other fields omitted
+  Seccomp *SeccompStrategyOptions `json:"seccomp,omitempty"`
+}
+
+// Defines the seccomp strategy type and any options used to create the strategy.
+type SeccompStrategyOptions struct {
+	// Rule is the strategy that will dictate the allowable seccomp profile name values that may be set.
+	Rule SeccompStrategy `json:"rule"`
+	// Profiles are a list of profile names that may be used.
+	Profiles []string `json:"profiles,omitempty"`
+}
+
+// SeccompStrategy denotes strategy types for generating SeccompProfileName values for a
+// SecurityContext.
+type SeccompStrategy string
+
+const (
+	// container must run as one of a set of profile names.
+	SeccompStrategyMustRunAs SeccompStrategy = "MustRunAs"
+	// container may make requests for any profile name.
+	SeccompStrategyRunAsAny SeccompStrategy = "RunAsAny"
+)
+```
+
+## Examples
+
+### Unconfined profile
+
+An unconfined profile can be implemented with a single default action:
+
+```yaml
+apiVersion: v1
+kind: SeccompProfile
+metadata:
+  name: unconfined
+defaultAction: allow
+```
+
+### Docker default profile
+
+We can model the docker default profile.  Profiles such as this should by
+convention contain the version number of the runtime they are for:
+
+```yaml
+apiVersion: v1
+kind: SeccompProfile
+metadata:
+  name: docker-v1-10
+defaultAction: kill
+rules:
+  - name: personality
```

Comment:
@jfrazelle I was originally going to keep the int values but I liked the strings better.  It should be possible to build a converter for existing seccomp profiles into this syntax.